### PR TITLE
Sync contacts with google - experimental

### PR DESCRIPTION
**Pull request by jacquarg**
A new konnector to allow contact sync between cozy and one google account: the contact of google are imported and merged in the cozy ; and then modifications on these contacts are synced between cozy and google. Futures adds in google adds in cozy. Contact deletion in cozy deletes in google ; deletion in google doesn't delete in cozy. 
Actually, profile picture sync doesn't work ; and be cautious with your contacts !

- add customView in konnector view, to offer more space to explain and provide tools.
- pass lastImport fields in konnector fetch.

### CODE REVIEW

**Comment by frankrousseau** [comment]
That one is tough to merge. We will look at it later.

**Comment by jacquarg** [comment]
Rebased on master. Development branch should be updated to master before reviewing for easier reading.

Some feature missings: 
* Some hack in the UI (javascript 'injection'), very specific to this konnector. Do somethings more generic ?
* No translation yet (depend on the question above)
* As in import from google, getting photo from google can be very long (hours) because of Google API Throttling.
* icon still missing

Ping @aenario , @frankrousseau

**Comment by frankrousseau** [comment]
Can you add google tags to contacts created from that import?

**Comment by jacquarg** [comment]
Yes ! 
Should I b e more explicit that it's not just an import, but a synchronization (including deletion of Google contacts on Cozy's contact deletion) ?

**Comment by frankrousseau** [comment]
The synchronization aspect scares me. I think it should just import data from Google.

**Comment by frankrousseau** [comment]
A least not push deletions.

**Comment by jacquarg** [comment]
Yes, it works, and the algorithm is basic, but it scares me to.
I can easily disconnect push to google functions, and we could create a special branch or something for warriors which want to test it extensively ?

**Comment by frankrousseau** [comment]
Here is what I propose: disable it and let it lives with the one way import. Then, we'll grab feedback and see if we meet bug. 
Once this first part validated. We will be able to add the push part.

File: server/konnectors/googlecontacts.coffee [hunk 1]
```diff
@@ -0,0 +1,342 @@
...
+    name: "Google Contacts"
+    slug: "googlecontacts"
+    description: "Synchronize google contacts with cozy through google's API. Experimental - please backup your contacts from your cozy and your google account."
+    vendorLink: "https://www.google.com/contacts/"
+
+    customView: """
+    <h6>Initialize or reset this konnector</h6>
+    <p>1. Press "connect your google account" button to connect to your Google account and authorize your Cozy to access to it. Google will provide you with a complex string. Once you get it copy it in your clipboard, we will use it in second step.</p>
+    <button id="connect-google" title="Connect your Google account" class="btn"
+       onclick="window.open('#{GoogleToken.getAuthUrl()}', 'Google OAuth', 'toolbars=0,width=700,height=600,left=200,top=200,scrollbars=1,resizable=1'); var input = $('#googlecontacts-authCode-input'); input.parents('.field').toggleClass('hidden'); input.attr('type', 'text'); input.val(''); $('#googlecontacts-accountName-input').val('');return false;"
```
Comment (jacquarg) [inline server/konnectors/googlecontacts.coffee]: Should we keep this hack ? Some js is mandatory (the window.open) , but most is just here for UX.
If yes, we will need a solution for translations.

**Comment by jacquarg** [comment]
@frankrousseau 
I deactivate the push to google functions. But I am stuck with the UI hack (... and translations)

**Comment by jacquarg** [comment]
@frankrousseau do you have a few minutes to speak about hacks in the UI (and then I will be able to finish this work)

**Comment by poupotte** [comment]
It's ok for me ! :)

**Comment by frankrousseau** [comment]
Thank you @jacquarg

**Pull request merged**